### PR TITLE
Update ESLint + Prettier

### PR DESCRIPTION
**Pull request by chaance**
- ESLint and all related dependencies updated
- Prettier updated
- Ran `prettier . --write` to reduce potential formatting issues in later commits

Unless I commented on a change, it came from `prettier . --write`.

### CODE REVIEW

File: docs/advanced-guides/migrating-5-to-6.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Migrating React Router v5 to v6
```
Comment (chaance) [inline docs/advanced-guides/migrating-5-to-6.md:1]: All changes here are from Prettier

File: docs/advanced-guides/migrating-reach-to-6.md [hunk 2]
```diff
@@ -96,7 +96,7 @@ function User(props) {
 }
```
Comment (chaance) [inline docs/advanced-guides/migrating-reach-to-6.md:96]: All changes here are from Prettier

File: docs/advanced-guides/testing/README.md [hunk 3]
```diff
@@ -8,4 +8,4 @@ We have guides for the following libraries:
 - [Mocha](testing-with-mocha.md)
```
Comment (chaance) [inline docs/advanced-guides/testing/README.md:8]: All changes here are from Prettier

File: docs/advanced-guides/working-with-the-search-string.md [hunk 4]
```diff
@@ -26,13 +26,13 @@ You can use the `useSearchParams` hook anywhere you need to work with the
 search/query parameters.
```
Comment (chaance) [inline docs/advanced-guides/working-with-the-search-string.md:26]: All changes here are from Prettier

File: docs/faq.md [hunk 5]
```diff
@@ -11,22 +11,17 @@ mean you can't use the router. Assuming you can actually use hooks (you're on
 React 16.8+), you just need a wrapper.
```
Comment (chaance) [inline docs/faq.md:11]: All changes here are from Prettier

File: docs/installation/add-to-a-website.md [hunk 6]
```diff
@@ -217,10 +217,10 @@ Go to the `index.js` file in your project and import the necessary functions fro
```
Comment (chaance) [inline docs/installation/add-to-a-website.md:217]: All changes here are from Prettier

File: docs/installation/getting-started.md [hunk 7]
```diff
@@ -18,18 +18,14 @@ Once you have React Router installed and running, come back and continue with th
```
Comment (chaance) [inline docs/installation/getting-started.md:18]: All changes here are from Prettier

File: docs/api-reference.md [hunk 8]
```diff
@@ -86,7 +86,7 @@ Sometimes you need to confirm navigation before it actually happens. For example
```
Comment (chaance) [inline docs/api-reference.md:86]: All changes here are from Prettier

File: package.json [hunk 9]
```diff
@@ -87,6 +87,7 @@
     }
   },
   "prettier": {
+    "arrowParens": "avoid",
```
Comment (chaance) [inline package.json:90]: This default changed from v1 to v2, so I added this preference to reduce the number of changes we have to worry about.

File: scripts/rollup/index.js [hunk 10]
```diff
@@ -2,7 +2,7 @@ import reactRouter from "./react-router.js";
 import reactRouterDOM from "./react-router-dom.js";
 import reactRouterNative from "./react-router-native.js";
 
-export default function (options) {
+export default function rollup(options) {
```
Comment (chaance) [inline scripts/rollup/index.js:5]: New react-app eslint config warns about anonymous default exports. Probably not a big deal for these files but this was simple enough to keep it happy!

**Review by mjackson — APPROVED** [review]
Looks great!

File: docs/advanced-guides/migrating-5-to-6.md [hunk 11]
```diff
@@ -73,11 +71,9 @@ function User({ id }) {
   <Route path="/about" component={About} />
   <Route
     path="/users/:id"
-    render={
-      ({ match }) => <User id={match.params.id} />
-    }
+    render={({ match }) => <User id={match.params.id} />}
   />
-</Switch>
+</Switch>;
```
[the remark this answers is not in the document]
Comment (chaance) [reply docs/advanced-guides/migrating-5-to-6.md:76]: We could put them in an `App` component. Seems like that would also have the benefit of adding context, no? Potential user confusion: "Wait, so does the Switch go in the User component or somewhere else?"

[the remark this answers is not in the document]
Comment (chaance) [reply docs/advanced-guides/migrating-5-to-6.md:76]:
```js
// v4 and v5 before 5.1
function User({ id }) {
  // ...
}

function App() {
  return (
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/about" component={About} />
      <Route
        path="/users/:id"
        render={({ match }) => <User id={match.params.id} />}
      />
    </Switch>
  );
}

// v5.1 preferred style
function User() {
  let { id } = useParams();
  // ...
}

function App() {
  return (
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>
      <Route path="/about">
        <About />
      </Route>
      {/* Can also use a named `children` prop */}
      <Route path="/users/:id" children={<User />} />
    </Switch>
  );
}
```

[the remark this answers is not in the document]
Comment (mjackson) [reply docs/advanced-guides/migrating-5-to-6.md:76]: Yeah, I like this 👍

**Pull request merged**